### PR TITLE
[SW BRC][HEVCe] Disable unnecessary checking in HDRConformance case

### PR DESCRIPTION
**Pull request by pgribov**
if NalHrdConformance is on and VuiNalHrdParameters is off (MFX_BRC_HRD_WEAK)
then there is no need to check some hrd parameters.
This checking was disabled for this case.

### CODE REVIEW

File: _studio/mfx_lib/shared/include/mfx_brc_common.h [hunk 1]
```diff
@@ -228,7 +243,7 @@ class cBRCParams
 public:
     cBRCParams() :
         rateControlMethod(0),
-        bHRDConformance(0),
+        HRDConformance(0),
```
Comment (mgonchar) [inline _studio/mfx_lib/shared/include/mfx_brc_common.h]: use ```MFX_BRC_NO_HRD``` here

File: _studio/mfx_lib/shared/include/mfx_brc_common.h [hunk 2]
```diff
@@ -160,6 +160,21 @@ class AVGBitrate
...
+--------------------------------------------------------------
+    off                  any                => MFX_BRC_NO_HRD
+    default              off                => MFX_BRC_NO_HRD
+    on                   off                => MFX_BRC_HRD_WEAK
+    on (or default)      on (or default)    => MFX_BRC_HRD_STRONG
+--------------------------------------------------------------
+*/
+enum : mfxU16
+{
+    MFX_BRC_NO_HRD,
```
Comment (mgonchar) [inline _studio/mfx_lib/shared/include/mfx_brc_common.h]: You are highly relying on default values start from zero, but not specifying it explicitly.  Better to add initialization

```MFX_BRC_NO_HRD = 0,```

File: _studio/mfx_lib/shared/src/mfx_brc_common.cpp [hunk 3]
```diff
@@ -153,14 +153,21 @@ mfxStatus cBRCParams::Init(mfxVideoParam* par, bool fieldMode)
...
+    if (pExtCO)
+    {
+        if (!IsOff(pExtCO->NalHrdConformance) && !IsOff(pExtCO->VuiNalHrdParameters))
+            HRDConformance = MFX_BRC_HRD_STRONG;
+        else if (IsOn(pExtCO->NalHrdConformance) && IsOff(pExtCO->VuiNalHrdParameters))
+            HRDConformance = MFX_BRC_HRD_WEAK;
+    }
 
-    if (bHRDConformance)
+    if (HRDConformance)
```
Comment (mgonchar) [inline _studio/mfx_lib/shared/src/mfx_brc_common.cpp]: Better to use explicit comparison with NO_HRD enum and not rely on defaults. Same applicable to many places below
```
if (HRDConformance != MFX_BRC_NO_HRD)
```

File: samples/sample_common/include/brc_routines.h [hunk 4]
```diff
@@ -43,12 +43,28 @@ or https://software.intel.com/en-us/media-client-solutions-support.
...
+--------------------------------------------------------------
+    off                  any                => MFX_BRC_NO_HRD
+    default              off                => MFX_BRC_NO_HRD
+    on                   off                => MFX_BRC_HRD_WEAK
+    on (or default)      on (or default)    => MFX_BRC_HRD_STRONG
+--------------------------------------------------------------
+*/
+enum : mfxU16
+{
+    MFX_BRC_NO_HRD,
```
Comment (mgonchar) [inline samples/sample_common/include/brc_routines.h]: same applicable for this file as well

File: _studio/mfx_lib/shared/include/mfx_brc_common.h (hunk 1 above)
Comment (pgribov) [reply _studio/mfx_lib/shared/include/mfx_brc_common.h]: changed

File: _studio/mfx_lib/shared/include/mfx_brc_common.h (hunk 2 above)
Comment (pgribov) [reply _studio/mfx_lib/shared/include/mfx_brc_common.h]: changed

File: _studio/mfx_lib/shared/src/mfx_brc_common.cpp (hunk 3 above)
Comment (pgribov) [reply _studio/mfx_lib/shared/src/mfx_brc_common.cpp]: changed

File: samples/sample_common/include/brc_routines.h (hunk 4 above)
Comment (pgribov) [reply samples/sample_common/include/brc_routines.h]: changed

**Pull request merged**